lib/gadget/gadget-glue.c

```diff
@@ -585,4 +609,22 @@ frida_objc_api_try_get (void)
   return api;
 }
 
+static gboolean
+frida_dylib_range_try_get (const char * apple[], GumMemoryRange * range)
+{
+  const char * entry;
+  int i = 0;
+
+  while ((entry = apple[i++]) != NULL)
+  {
+    if (!strncmp (entry, "frida_dylib_range=", 18))
+    {
+      if (sscanf (entry, "frida_dylib_range=0x" GUM_PFMT ",0x%lx", &range->base_address, &range->size) == 2)
```

Comment:
Using `G_GINT64_MODIFIER` and `G_GSIZE_MODIFIER` here would be more precise and portable.